### ISSUE TITLE
Can't get right-aligned colvis dropdown

### ISSUE THREAD

**Issue by miken32** [DataTables/Buttons#140]
According to [latest Bootstrap 4 documentation](https://getbootstrap.com/docs/4.0/components/dropdowns/#menu-alignment), I should be able to add `.dropdown-menu-right` to the existing `div.dropdown-menu` to get a right-aligned menu, but this is not working. I suspect something in DT is overriding the Bootstrap positioning code (positioning for Bootstrap dropdowns is done by popper.js, not CSS.)

I simply add this before creating the table with buttons:

    $.fn.dataTable.Buttons.defaults.dom.collection.className += " dropdown-menu-right";

And I see the class appears on the element as expected, but there's no positioning change.

**Comment by DataTables** [comment]
Yes [Buttons does its own positioning](https://github.com/DataTables/Buttons/blob/6072abad6648d90e5248ac1ba8f6cdaaaaa97192/js/dataTables.buttons.js#L1197).

**Comment by miken32** [comment]
Doesn't look like there's an easy way to override that without rewriting the whole `action` function? It would be great if the code in `buttons.bootstrap4.js` could restore the default Bootstrap behaviour.

**Comment by DataTables** [comment]
That's correct - at the moment there isn't a way to override that positioning. I will look into options for using the Bootstrap specific code to restore the Bootstrap behaviour though.

**Comment by miken32** [comment]
I look forward to it. I'd have a try myself, but this is way out of my depth!

**Comment by kinguru** [comment]
Is there a documented option how to make dropdown menu aligned right?
"dropdown-menu-right"

**Comment by AllanJard** [comment]
https://datatables.net/reference/button/collection - `align: 'button-right'` is probably what you want now. If that doesn't help please open a thread on the forums with a link to a test case.
